code/modules/shuttle/shuttle.dm

```diff
@@ -801,6 +801,10 @@
 			if(!QDELETED(E))
 				engine_list += E
 				. += E.engine_power
+		for(var/obj/machinery/shuttle/engine/E in areaInstance.contents)
```

Comment:
Notice the for loop directly above this one that also runs.

Reply:
One is for the structure engines (the current ones which are just scenery) and the other is for my machine thrusters which are buildable by the players

Reply:
One is for the structure engines (the current ones which are just scenery) and the other is for my machine thrusters which are buildable by the players

Reply:
Yeah but currently:

1. Both for loops run through the same list with the same type
2. Both are adding different values to the value that gets returned. All of the `engine_power` for all of the engines gets added as well as a `1` for every active thruster.
3. Every engine is getting added to the `engine_list` twice because you're just looping through all of the engines in two separate for loops.

Reply:
They are of different types one is:
var/obj/structure/shuttle/engine/E the other is 
var/obj/machinery/shuttle/engine/E

Reply:
My bad, I'm blind.